Userland/Applications/PixelPaint/Tools/GradientTool.cpp

```diff
@@ -0,0 +1,357 @@
+    auto t_gradient_begin_line = Gfx::FloatLine(as_translated(m_gradient_begin_line.a()), as_translated(m_gradient_begin_line.b()));
+    auto t_gradient_center_line = Gfx::FloatLine(as_translated(m_gradient_center_line.a()), as_translated(m_gradient_center_line.b()));
+    auto t_gradient_end_line = Gfx::FloatLine(as_translated(m_gradient_end_line.a()), as_translated(m_gradient_end_line.b()));
+    int width = m_editor->active_layer()->rect().width() * scale;
+    int height = m_editor->active_layer()->rect().height() * scale;
+
+    float rotation_radians = atan2f(t_gradient_begin_line.a().y() - t_gradient_end_line.a().y(), t_gradient_begin_line.a().x() - t_gradient_end_line.a().x());
+    float rotation_degrees = ((rotation_radians * 180) / static_cast<float>(M_PI)) - 90;
+
+    auto scaled_gradient_center = m_gradient_center.value().to_type<float>().scaled(scale, scale).to_type<int>();
+    auto gradient_rect_height = 2 * (scaled_gradient_center.y() < 0 ? AK::abs(scaled_gradient_center.y()) + height : scaled_gradient_center.y() > height ? scaled_gradient_center.y()
+                                                                                                                                                         : AK::max(scaled_gradient_center.y(), height - scaled_gradient_center.y()));
+    auto gradient_rect_width = 2 * (scaled_gradient_center.x() < 0 ? AK::abs(scaled_gradient_center.x()) + width : scaled_gradient_center.x() > width ? scaled_gradient_center.x()
+                                                                                                                                                      : AK::max(scaled_gradient_center.x(), width - scaled_gradient_center.x()));
+    auto gradient_max_side_length = AK::max(gradient_rect_height, gradient_rect_width);
+    auto t_gradient_center = static_cast<Gfx::IntPoint>(as_translated({ m_gradient_center.value().x(), m_gradient_center.value().y() }));
+    auto gradient_rect = Gfx::IntRect::centered_at(t_gradient_center, { gradient_max_side_length, gradient_max_side_length });
+    float overall_gradient_length_in_rect = calculate_gradient_length(gradient_rect.size(), rotation_degrees);
+
+    if (m_gradient_half_length == 0 || overall_gradient_length_in_rect == 0 || overall_gradient_length_in_rect == AK::NaN<float>)
+        return;
+
+    auto gradient_half_width_percentage_offset = (m_gradient_half_length * scale) / overall_gradient_length_in_rect;
+    auto color_to_use = m_editor->color_for(GUI::MouseButton::Primary);
+    int base_opacity = color_to_use.alpha() * m_opacity / 100;
+    color_to_use.set_alpha(base_opacity);
+    auto gradient_start_color = color_to_use;
+    gradient_start_color.set_alpha(0);
+
+    painter.fill_rect_with_linear_gradient(gradient_rect, Array { Gfx::ColorStop { gradient_start_color, 0.5f - gradient_half_width_percentage_offset }, Gfx::ColorStop { color_to_use, 0.5f + gradient_half_width_percentage_offset } }, rotation_degrees);
```

Comment:
Well that's just how i would like to use it :) My plan was to have a symmetric gradient  on both sides of the middle line (middle should be 50% of the gradient).
For the future it would be great to have an alternative mode where both sides could have a different length and therefore have a smooth or hard start/end of the gradient.